version: "3.8"

services:
  mongo_db:
    container_name: mongo_db
    image: mongo:4.4.2-bionic
    restart: always
    env_file:
      - .env
    volumes:
      - mongo_data:/data/db
    ports:
      - 27017:27017

  data_receiver:
    container_name: data_receiver
    image: gr1902/data_manager
    build:
      context: .
      dockerfile: Dockerfile
    env_file:
      - .env
    command: python start_receiver.py --config config.yaml
    volumes:
      - ./logs/:/home/user/app/logs
    restart: on-failure
    depends_on:
      - mongo_db

  data_checker:
    container_name: data_checker
    image: gr1902/data_manager
    build:
      context: .
      dockerfile: Dockerfile
    env_file:
      - .env
    command: python start_checker.py --config config.yaml
    volumes:
      - ./logs/:/home/user/app/logs
    restart: on-failure
    depends_on:
      - mongo_db

volumes:
  mongo_data: